src/controllers/enter.information.controller.ts

```diff
@@ -86,8 +87,11 @@ const getObjectionFromSession = async (req: Request) => {
     throw new Error("No session present");
   }
   const objection: Objection = await getObjection(req.session);
+  const companyProfileInSession: ObjectionCompanyProfile = retrieveFromObjectionSession(req.session, SESSION_COMPANY_PROFILE);
+  if (companyProfileInSession) {
+    addToObjectionSession(req.session, PREVIOUSLY_SELECTED_COMPANY, companyProfileInSession.companyNumber);
+  }
   if (!objection) {
```

Comment:
Do we still need this if check?

Reply:
In a normal circumstance, it works either way, I guess it's just a fail safe in case it tries to add something undefined into the previously selected company value if company profile were for some reason not present.